src/components/CalendarContainer.ts

```diff
@@ -0,0 +1,526 @@
+        context.setContext(mxObject.getEntity(), mxObject.getGuid());
+        if (onDropEvent === "callMicroflow" && onDropMicroflow && mxObject.getGuid()) {
+            window.mx.ui.action(onDropMicroflow, {
+                context,
+                origin: mxform,
+                error: error => window.mx.ui.error(
+                    `Error while executing microflow: ${onDropMicroflow}: ${error.message}`
+                )
+            });
+        } else if (onDropEvent === "callNanoflow" && onDropNanoflow.nanoflow) {
+            window.mx.data.callNanoflow({
+                nanoflow: onDropNanoflow,
+                origin: mxform,
+                context,
+                error: error => window.mx.ui.error(
+                    `An error occurred while executing the nanoflow: ${error.message}`
+                )
+            });
+        }
+    }
+
+    public static validateProps(props: CalendarContainerProps): ReactChild {
+        const errorMessage: string[] = [];
+
+        if (props.onClickEvent === "callMicroflow" && !props.eventMicroflow) {
+            errorMessage.push("On click event is set to 'Call a microflow' but no microflow is selected");
+        } else if (props.onClickEvent === "callNanoflow" && !props.eventNanoflow.nanoflow) {
+            errorMessage.push("On click event is set to 'Call a nanoflow' but no nanoflow is selected");
+        }
+        if (props.onClickSlotEvent === "callMicroflow" && !props.slotMicroflow) {
+            errorMessage.push("On click slot is set to 'Call a microflow' but no microflow is selected");
+        } else if (props.onClickSlotEvent === "callNanoflow" && !props.slotNanoflow.nanoflow) {
+            errorMessage.push("On click slot is set to 'Call a nanoflow' but no nanoflow is selected");
+        }
+        if (props.onDropEvent === "callMicroflow" && !props.onDropMicroflow) {
+            errorMessage.push("On drop event is set to 'Call a microflow' but no microflow is selected");
+        } else if (props.onDropEvent === "callNanoflow" && !props.onDropNanoflow.nanoflow) {
+            errorMessage.push("On drop event is set to 'Call a nanoflow' but no nanoflow is selected");
+        }
+        if (props.onResizeEvent === "callMicroflow" && !props.onResizeMicroflow) {
+            errorMessage.push("On resize event is set to 'Call a microflow' but no microflow is selected");
+        } else if (props.onResizeEvent === "callNanoflow" && !props.onResizeNanoflow.nanoflow) {
+            errorMessage.push("On resize event is set to 'Call a nanoflow' but no nanoflow is selected");
+        }
+        if (errorMessage) {
+            errorMessage.push(`Error in calendar configuration: ${errorMessage}`);
+        }
+        if (props.dataSource === "context" && (props.mxObject && props.mxObject.getEntity() !== props.eventEntity)) {
+            errorMessage.push(`${props.friendlyId}: Context entity does not match the event entity`);
+        }
+
```

Comment:
Should we also check if the objects are editable... and if we can create an object,
Maybe show a console warning and disabled the function...